tests/integrationv2/test_serialization.py

```diff
@@ -0,0 +1,151 @@
+import pytest
+import copy
+import os
+import random
+import string
+
+from configuration import available_ports
+from common import ProviderOptions, Protocols, data_bytes
+from fixtures import managed_process  # lgtm [py/unused-import]
+from providers import Provider, S2N
+from utils import invalid_test_parameters, get_parameter_name, to_bytes
+
+SERVER_STATE_FILE = 'server_state'
+CLIENT_STATE_FILE = 'client_state'
+
+SERVER_DATA = f"Some random data from the server:" + "".join(
+    random.choice(string.ascii_uppercase + string.digits) for _ in range(10)
+)
+CLIENT_DATA = f"Some random data from the client:" + "".join(
+    random.choice(string.ascii_uppercase + string.digits) for _ in range(10)
+)
+
+"""
+This test file checks that a serialized connection can be deserialized by an older version of
+s2n-tls and vice versa. This ensures that any future changes we make to the handshake are backwards-compatible
+with an older version of s2n-tls.
+
+This feature requires an uninterrupted TCP connection with the peer in-between serialization and
+deserialization. Our integration test setup can't provide that while also using two different s2n-tls
+versions. To get around that we do a hack and serialize/deserialize both peers in the TLS connection.
+This prevents one peer from receiving a TCP FIN message and shutting the connection down early.
+"""
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("protocol", [Protocols.TLS13, Protocols.TLS12], ids=get_parameter_name)
+def test_serialize_new_deserialize_old(managed_process, tmp_path, protocol):
```

Comment:
I think I described this poorly: https://github.com/aws/s2n-tls/pull/4548#discussion_r1595472635

This was the suggestion:
```suggestion
@pytest.mark.parametrize("use_mainline_version", [True, False], ids=get_parameter_name)
def test_serialize_new_deserialize_old(managed_process, tmp_path, protocol, use_mainline_version):
```

I don't think it necessitates using something in another test.

Reply:
Ah I think I get it. I'll try it out. I'm worried making this a parameter will make it harder to understand the test, but you can lmk if it's any more or less confusing.

Reply:
I think I like it better now because of all the duplicated code that can be removed, but you or others disagree feel free to put it back!